### ISSUE TITLE
submitted text files that produce socket error

### ISSUE THREAD

**Issue by bill-auger** [saasbook/hw-sinatra-saas-hangperson#20]
if you have submitted a text file that produces "socket error" please drag and drop that file to a comment here so we can test it

**Comment by edxdebugging3** [comment]
[file2.txt](https://github.com/saasbook/hw-sinatra-saas-hangperson/files/325158/file2.txt)

**Comment by bill-auger** [comment]
NOTE: TO ADMINS:

im not sure if this is the only source of this issue but it seems like this caveat could be fixed with a regex in the grader side to strip the protocol bits

**Comment by tansaku** [comment]
thanks @bill-auger - this would be a great issue to get sorted ...

**Comment by bill-auger** [comment]
@edxdebugging3 

see the note above

**Comment by peeyushjh** [comment]
[Hw_2_Sinatra_Hangperson.txt](https://github.com/saasbook/hw-sinatra-saas-hangperson/files/343595/Hw_2_Sinatra_Hangperson.txt)


Not able to grade the file. Nothing is happening after the screen in the snapshot, after I Check for the grader
![screen shot 2016-07-01 at 7 23 23 pm](https://cloud.githubusercontent.com/assets/10037878/16523941/94059ffe-3fc3-11e6-80b8-66e718364a88.png)

**Comment by bill-auger** [comment]
@peeyushjh

which program did you use to create that file?

it was loaded with some crazy unicode

in the terminal it looks like this:
```
$ cat Hw_2_Sinatra_Hangperson.txt
��stormy-brushlands-58894.herokuapp.com
```
in nano that file looks like this:
```
$ nano Hw_2_Sinatra_Hangperson.txt
 ��s^@t^@o^@r^@m^@y^@-^@b^@r^@u^@s^@h^@l^@a^@n^@d^@s^@-^@5^@8^@8^@9^@4^@.^@h^@e^@r^@o^@k^@u^@a^@p^@p^@.^$
```

**Comment by RobertStroud** [comment]
@peeyushjh @bill-auger - the file appears to be UTF-16, little endian...

> $ od -c Hw_2_Sinatra_Hangperson.txt 
0000000  377 376   s  \0   t  \0   o  \0   r  \0   m  \0   y  \0   -  \0
0000020    b  \0   r  \0   u  \0   s  \0   h  \0   l  \0   a  \0   n  \0
0000040    d  \0   s  \0   -  \0   5  \0   8  \0   8  \0   9  \0   4  \0
0000060    .  \0   h  \0   e  \0   r  \0   o  \0   k  \0   u  \0   a  \0
0000100    p  \0   p  \0   .  \0   c  \0   o  \0   m  \0 
0000114
$ file Hw_2_Sinatra_Hangperson.txt 
Hw_2_Sinatra_Hangperson.txt: Little-endian UTF-16 Unicode text, with no line terminators
$

**Comment by RobertStroud** [comment]
NOTE: TO ADMINS:

The autograder should be more forgiving about the input it accepts - ideally, it should accept either a hostname or a URL and not be fussy about whitespace. Alternatively, it should give a sensible error message if it isn't given exactly what it needs.

Similarly, it shouldn't hang if the input is encoded in UTF-16 rather than ASCII.

**Comment by peeyushjh** [comment]
Thanks @bill-auger ... I got it

**Comment by RobertStroud** [comment]
NOTE: TO ADMINS:

Like many other people, I had trouble submitting HW2. I think the instructions are misleading and the submission method is fragile

- My first attempt had trailing white space - text files should always terminate with a newline
- My second attempt include "https:" and the trailing slash (it was a URL)
- My third attempt worked - it was just the host name for the Heroku server

By definition, a URL must always begin with a "scheme:" prefix - hence "foo.herokuapp.com" is not a URL. In other words, the instructions are technically incorrect, even though they tell you what to do.

Moreover, the autograder should be a bit more robust and forgiving about the format of the submission.

Why do we need to submit a text file at all? Why not simply paste a URL into a text box?

Thanks.

**Comment by bill-auger** [comment]
@RobertStroud 

i do agree with you that there should be a more forgiving regex - unfortunately this was somewhat of a counter-example - namely because he was clearly not using the recommended environment for the course - the submission courseware page clearly says to create this file with echo

**Comment by RobertStroud** [comment]
@bill-auger - I agree that the Unicode problem that @peeyushjh experienced is unusual. It's also true that the instructions are clear, apart from the confusion between URL and hostnames. But I do think the autograder should be more robust and should be willing to accept text files with trailing newlines, and URLs as well as hostnames.

**Comment by tatyana12** [comment]
https://berk1.herokuapp.com/new

**Comment by tatyana12** [comment]
[berk1.txt](https://github.com/saasbook/hw-sinatra-saas-hangperson/files/346898/berk1.txt)

**Comment by bill-auger** [comment]
NOTE TO ADMINS:

 tatyana12's file is UTF8 and contain this text:

`–n “myapp.herokuapp.com”`

but curiously this does not make the grader choke but completes successfully with 0/80 grade

**Comment by bill-auger** [comment]
NOTE TO ADMINS:

@peeyushjh s file is UTF-16 and causes the grader to hang indefinitely

**Comment by tansaku** [comment]
@RobertStroud yeah - I wish we had a textbox only submission - it should be possible in principle, but something about the autograder submission assumes that a file is being submitted.

I did spend a couple of days trying to create an edX form in 2015 that would allow submission directly to the grader, but failed :-(

Maybe time to try it again ...
